File: client/lib/core/manila_day.dart
/// `YYYY-MM-DD` in Manila for [instant].
///
/// The day the server calls "today": every database connection runs at
/// +08:00, so CURDATE() turns over at midnight in Manila. Computed from the
/// instant rather than the device's own zone, so a phone set to another zone
/// still agrees with the server about which day it is. The Philippines keeps
/// no daylight saving, so the fixed offset is always exact.
String manilaDayOf(DateTime instant) {
  final manila = instant.toUtc().add(const Duration(hours: 8));
  String pad(int n) => n.toString().padLeft(2, '0');
  return '${manila.year}-${pad(manila.month)}-${pad(manila.day)}';
}

File: server/src/lib/manila-day.js
'use strict';

const MANILA_OFFSET_MS = 8 * 60 * 60 * 1000;

/// YYYY-MM-DD in Manila, `daysBack` days before `now`.
///
/// The same day MySQL's CURDATE() gives, now that every pooled connection
/// runs at +08:00 (see db/pool.js), so a caption built here agrees with data
/// windowed there. Fixed-offset arithmetic rather than the host's local
/// getters: those follow whatever zone the server machine is set to. The
/// Philippines keeps no daylight saving, so +08:00 is always exact.
function manilaDay(daysBack = 0, now = new Date()) {
  const d = new Date(now.getTime() + MANILA_OFFSET_MS);
  d.setUTCDate(d.getUTCDate() - daysBack);
  return d.toISOString().slice(0, 10);
}

module.exports = { manilaDay };
